paddle/phi/kernels/funcs/selected_rows_functor.cc

```diff
@@ -371,7 +371,14 @@ add_sparse_inputs(const std::vector<const phi::SelectedRows*>& inputs,
     auto& input_rows = input->rows();
 
 #ifdef PADDLE_WITH_MKLDNN
-    paddle::operators::OneDNNAXPYHandler<T> axpy_handler(input_width, T(1.f));
+
+    auto& pool = paddle::platform::DeviceContextPool::Instance();
```

Comment:
We don't recommend using `paddle::platform::DeviceContextPool::Instance()` in PHI. You could add `DeviceContext`  argument for this function to pass `dev_ctx`

Reply:
Thanks for catching this.
This function has `CPUContext` instead of `OneDNNContext` so I'm unable to use `GetDevice()` method. I removed fluid dependency and created `OneDNNContext` before this function is called. Could you please advice if this approach https://github.com/PaddlePaddle/Paddle/pull/47779/commits/0c9ca315231e3ab4d272f5a629a76e7642f5e267 is good?

Reply:
It seems good.